src/kakarot/evm.cairo

```diff
@@ -278,3 +287,32 @@ namespace EVM {
         );
     }
 }
+
+namespace Internals {
+    func is_jumpdest_valid{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        valid_jumpdests: DictAccess*,
+    }(code_address: felt, is_create: felt, index: felt) -> felt {
+        alloc_locals;
+        let (is_cached) = dict_read{dict_ptr=valid_jumpdests}(index);
+        let is_create_or_cached = is_create + is_cached;
+
+        // If a create message, the account is not deployed on SN, so we just return early
+        // the value cached is correct as we cache all valid jumpdests on initcodes deployment.
+        if (is_create_or_cached != 0) {
+            return is_cached;
+        }
+
+        let code_starknet_address = Account.compute_starknet_address(code_address);
+        let (is_valid) = IAccount.is_jumpdest_valid(code_starknet_address, index);
+
+        if (is_valid == 0) {
+            return FALSE;
+        }
+
+        dict_write{dict_ptr=valid_jumpdests}(index, TRUE);
+        return TRUE;
```

Comment:
so this is

```
dict_write{dict_ptr=valid_jumpdests}(index, is_valid);

return is_valid;
```

and you remove the if/else at the cost of writing 0 while it's not necessary, meaning that you save some steps for valid cases and use some more for invalid. I think it's then better